pkg/store/storepb/rpc.proto

```diff
@@ -35,6 +35,18 @@ message SeriesRequest {
   int64 min_time                 = 1;
   int64 max_time                 = 2;
   repeated LabelMatcher matchers = 3 [(gogoproto.nullable) = false];
+
+  int64 maxAggregateWindow = 4;
```

Comment:
Hmm.. Can we change this name? You cannot tell from `maxAggregateWindow` what unit of this is it.. Is window in samples? or seconds? Or hours? We were talking about 5m, 1h windows etc so I assume timestamp here, but it is not so clear from the var name. Also it looks like API requires exact window time (for example if you pass 5m and 23 sec it will return invalid arg). What about changing it to `resolution`? From the query API you can tell that this is time and that you don't really need to pick one from existing. 

And what if we maybe want to pick some defined windows? If that's true maybe we should name this var `window` and give it `enum` type?

Reply:
can see need for `max` prefix, because some window can be unavailable , but `aggregate` word seems not needed

Reply:
Strictly speaking, there is no unit attached to the window. Timestamps in TSDB are just numbers. Just like the query range only implicitly is "milliseconds since epoch". Of course the canonical choice for data produced from Prometheus is `millisecond` – but there's basically nothing to be gained from forcing it. Especially because none of the underlying APIs do.

Maybe `maxResolution` is better indeed.

Where are we erroring right now if it's not a predefined one? I'd regardless not enforce those at an API level. It adds a constraint on reusability and all clients would in the end compute their arbitrary number and then ceil or floor it to the fixed windows. We might as well just do that at the backend for everyone.

Reply:
> Strictly speaking, there is no unit attached to the window.

Sorry for the confusion, maybe "unit" is not the good wording. For me "this is timestamp" statement is sufficient enough to know what to put in this var as a user. "Window" was just too ambiguous. In the same way, we have `max_time`, not `max`. `max_resolution` maybe be fine, since promQL resolution is time as well... but one could argue that it should be `max_resolution_time`. I am fine with the shorter.

> Where are we erroring right now if it's not a predefined one? I'd regardless not enforce those at an API level. It adds a constraint on reusability and all clients would, in the end, compute their arbitrary number and then ceil or floor it to the fixed windows. We might as well just do that at the backend for everyone.

I think you misunderstood me. I totally agree with you. I only said:
> Also it looks like API requires exact window time 

We are not erroring anywhere right now, but API suggests that by saying "window", but maybe that is only mine perception here (:

Also: Remember to not use _camel case_ here, just good old _snake case_ in proto (: (Missed that totally in review)

Reply:
Great, will move to `max_resolution` then – and also update all other namings of `window` I suppose :)

Reply:
yea